### PR TITLE
Run edmEventSize on root files produced by igprof workflow and save to compare with same profiling PR's

### PR DESCRIPTION
**Pull request by gartung**
Pulls in changes from #1408

### CODE REVIEW

**Comment by gartung** [comment]
@jpata @smuzaffar

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
please test

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
enable profiling

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@jpata , can you please check the comparison logs here and see every thing ok? let me know if this is good to go in now

**Comment by jpata** [comment]
I was looking through the outputs and I can find the product comparisons for 136.874.

However, I cannot find the profiling outputs, so I cannot determine if profiling was run and if the comparison outputs are stored also for profiling workflows.

**Comment by gartung** [comment]
The profiling outputs for the IB are stored in 
https://cmssdt.cern.ch/SDT/jenkins-artifacts/igprof/CMSSW_11_2_X_2020-11-12-2300/slc7_amd64_gcc820/profiling/23434.21/

**Comment by gartung** [comment]
@smuzaffar profiling needs to be enabled for the comparisons to be made. I am not a level 2 for this repo so my comment above was ignored by cms-bot.

File: pr_testing/test_multiple_prs.sh [hunk 1]
```diff
@@ -1196,9 +1196,18 @@ for BT in ${ENABLE_BOT_TESTS}; do
              echo "<li><a href=\"$PROFILING_WORKFLOW/\">$PROFILING_WORKFLOW/</a> </li>" >> $WORKSPACE/upload/profiling/index.html
              get_jenkins_artifacts igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/RES_CPU_step3.txt  ${CMSSW_IB}_RES_CPU_step3.txt || true
              $WORKSPACE/profiling/Analyze_tool/compare_cpu_txt.py --old ${CMSSW_IB}_RES_CPU_step3.txt --new RES_CPU_step3.txt > RES_CPU_compare_$PROFILING_WORKFLOW.txt || true
-             mkdir -p $WORKSPACE/upload/profiling/
-             cp -p RES_CPU_compare_$PROFILING_WORKFLOW.txt $WORKSPACE/upload/profiling/$d/ || true
              echo "<li><a href=\"$PROFILING_WORKFLOW/RES_CPU_compare_$PROFILING_WORKFLOW.txt\">Igprof Comparison cpu usage RECO produce methods.</a> </li>" >> $WORKSPACE/upload/profiling/index.html
+             cp $WORKSPACE/cms-bot/comparisons/compareProducts.* ./
+             if (get_jenkins_artifacts igprof/${CMSSW_IB}/${ARCHITECTURE}/profiling/${PROFILING_WORKFLOW}/step3_sizes_${PROFILING_WORKFLOW}.txt  ${CMSSW_IB}_step3_sizes_${PROFILING_WORKFLOW}.txt) ; then
+               edmEventSize -v step3*.root > step3_sizes_${PROFILING_WORKFLOW}.txt
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:1202]: is there any chance that this command might fail? If yes then add `|| true` at the end so that it does not break the whole comparison test

Comment (gartung) [reply pr_testing/test_multiple_prs.sh]: Done.

File: comparisons/compareProducts.sh [hunk 2]
```diff
@@ -45,7 +45,7 @@ case `basename $fA` in
 
     rm ${os} ${ns}
   ;;
-  step*.txt)
+  *step*.txt)
```
Comment (smuzaffar) [inline comparisons/compareProducts.sh:48]: is this change on purpose?

Comment (gartung) [reply comparisons/compareProducts.sh:48]: Yes. The old file has the IB name prepended to it.

Comment (gartung) [reply comparisons/compareProducts.sh:48]: I can change that.

Comment (smuzaffar) [reply comparisons/compareProducts.sh:48]: no, I was not sure if it is on purpose or just a typo :-)

**Comment by smuzaffar** [comment]
test parameters:
  enable_test = profiling

**Comment by smuzaffar** [comment]
please test
not sure if bot is going to enable profiling for thsi repo or not :-) but lets try

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
test

**Comment by jpata** [comment]
The standard workflow size comparison looks good.

The profiling job was run and the size txt files exist, but the size comparison for profiling seems empty: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-823602/10766/profiling/23434.21/products_AOD_sizes_compare_23434.21.txt

I think there should be some message/header printed there despite there not being any reco size difference. Possibly there might be a space missing before the redirect symbol where the comparison script is called? https://github.com/cms-sw/cms-bot/pull/1413/files#diff-5e02a17340b02f75de80cf0d3a45e66970f1884b0a5cdf726ef1223846c95af3R1203-R1211

```bash
./compareProducts.sh ... _ 100 10> out.txt
```

**Comment by gartung** [comment]
The first try of edmEventSize produced an empty file. I think this was because the root files were not written to the current wirking diretory. I tried changing the root file location to $WORKSPACE/step3.root. I can't see if the output was empty this time since cmssdt is slow to respond at the moment.

**Comment by gartung** [comment]
I see what you mean now. I added a space before the redirect.

**Comment by gartung** [comment]
test

**Comment by gartung** [comment]
> I can't see if the output was empty this time since cmssdt is slow to respond at the moment.

Doh! The file is empty so there was nothing to show.

**Comment by jpata** [comment]
Thank you for the updates Patrick, now it looks good!

- [x] the event sizes for the standard workflow 136.874 get compared, output is in `Comparison with the baseline -> products_AOD.log/products_MINIAOD.log`
- [x] the event sizes for the profiling workflow get compared, output is in `Profiling results -> edmEventSize Comparison AOD/miniAOD output`

I think for reco PR reviews, this will certainly be very helpful! From my side it looks good to go, @slava77 any comments/suggestions?

Maybe something to consider would be a warning message in Comparison Summary if e.g. the standard workflow 136.874 has a size difference bigger than some threshold in AOD/MINIAOD, although this would be a "nice-to-have" rather than a necessity IMO.

**Comment by gartung** [comment]
@smuzaffar Can you please merge.

**Comment by smuzaffar** [comment]
+externals
